lib/ui-element.ts

```diff
@@ -474,35 +453,161 @@ export class UIElement {
     * @param direction
     */
     public async swipe(direction: Direction) {
-        const rectangle = await this.getRectangle();
-        const centerX = rectangle.x + rectangle.width / 2;
-        const centerY = rectangle.y + rectangle.height / 2;
-        let swipeX;
-        if (direction == Direction.right) {
-            const windowSize = await this._driver.getWindowSize();
-            swipeX = windowSize.width - 10;
-        } else if (direction == Direction.left) {
-            swipeX = 10;
+        logInfo(`Swipe direction: `, Direction[direction]);
+        if (this._args.isIOS) {
+            await this._driver
+                .execute('mobile: scroll', {
+                    element: <any>this._element.value,
+                    direction: Direction[direction]
+                });
         } else {
-            console.log("Provided direction must be left or right !");
+            try {
+                await this.scroll(direction);
+            } catch (error) {
+                console.log("", error);
+            }
         }
+        logInfo(`End swipe`);
+    }
+
+    /**
+    * Drag element with specific offset
+    * @experimental
+    * @param direction
+    * @param yOffset 
+    * @param xOffset - default value 0
+    */
+    public async drag(direction: Direction, yOffset: number, xOffset: number = 0, duration?: number) {
+        direction = direction === Direction.up ? Direction.down : Direction.up;
+
+        const location = await this.location();
+
+        const x = location.x === 0 ? 10 : location.x;
+        const y = location.y === 0 ? 10 : location.y;
+
+        const endPoint = calculateOffset(direction, y, yOffset, x, xOffset, this._args.isIOS);
+        duration = duration || endPoint.duration;
 
         if (this._args.isAndroid) {
-            const action = new this._wd.TouchAction(this._driver);
-            action.press({ x: centerX, y: centerY })
-                .wait(200)
-                .moveTo({ x: swipeX, y: centerY })
+            let action = new this._wd.TouchAction(this._driver);
+            action
+                .longPress({ x: x, y: y })
```

Comment:
Sadly in some cases longPress doesn't work. Example: RadListView Reorder with handle